vividus-plugin-rest-api/src/main/java/org/vividus/bdd/steps/api/HttpRequestSteps.java

```diff
@@ -110,12 +110,22 @@ public void putMultipartRequest(ExamplesTable requestParts)
     @When("I set request headers:$headers")
     public void setUpRequestHeaders(ExamplesTable headers)
     {
-        List<Header> requestHeaders = headers.getRowsAsParameters(true).stream()
-                .map(row -> new BasicHeader(row.valueAs(NAME, String.class), row.valueAs(VALUE, String.class)))
-                .collect(toList());
+        List<Header> requestHeaders = getRequestHeaders(headers);
         httpTestContext.putRequestHeaders(requestHeaders);
     }
 
+    /**
+     * Add request headers
+     * @param headers ExamplesTable representing list of headers with columns "name" and "value" specifying HTTP header
+     * names and values respectively
+     */
+    @When("I add request headers:$headers")
+    public void addRequestHeaders(ExamplesTable headers)
+    {
+        List<Header> requestHeaders = getRequestHeaders(headers);
+        httpTestContext.addRequestHeaders(requestHeaders);
```

Comment:
maybe create function like
```
void performWithHeaders(ExamplesTable headers, Consumer<List<Header>> headersConsumer) 
```
to inline setUpRequestHeaders and addRequestHeaders methods bodies